models/base_gan_model.py

```diff
@@ -110,9 +115,14 @@ def __init__(self, opt, rank):
 
         if "sam" in opt.D_netDs:
             self.use_sam = True
-            self.netfreeze_sam, self.predictor_sam = load_sam_weight(
-                self.opt.D_weight_sam
-            )
+            if opt.model_type_sam == "sam":
```

Comment:
Same remark as for diffusion, this should go to sam_inference.py as much as possible.